### PR TITLE
Web-components: Custom Elements Manifest v1 support

### PR DESCRIPTION
**Pull request by thepassle**
Issue:

## What I did
This adds support for the V1 version of the Custom Elements Manifest schema. This change is non breaking

TODO:
- I'd like to test the changes locally, but I havent figured out how to make that happen (as mentioned on discord)

## How to test

- Is this testable with Jest or Chromatic screenshots?
- Does this need a new example in the kitchen sink apps?
- Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

**Comment by gaetanmaisse** [comment]
@thepassle feels free to ping me on discord if you need help I would be glad to try to help 😉

File: examples/web-components-kitchen-sink/.storybook/preview.js [hunk 1]
```diff
@@ -1,10 +1,31 @@
+/* eslint-disable */
 /* global window */
 
-import { addParameters, setCustomElements } from '@storybook/web-components';
+import {
+  addParameters,
+
+  // setCustomElementsManifest,
+  // ☝️☝️☝️☝️☝️☝️
+  // @TODO: cant import this function for some reason after running `yarn bootstrap --core`
```
Comment (thepassle) [inline examples/web-components-kitchen-sink/.storybook/preview.js:9]: Hi @shilman , still havent managed to figure this out unfortunately ☹️ 

I have:
- Made my changes locally to: `addons/docs/src/frameworks/web-components/custom-elements.ts` and `app/web-components/src/client/customElements.ts`
- run `yarn bootstrap --core`
- run `cd examples/web-components-kitchen-sink/`
- run `yarn storybook`

But for some reason its giving me this error:
image

I have been able to test my changes by just copy pasting the function into `preview.js` and trying it out, but (if I understand correctly) I'd expect the function to be exposed from `@storybook/webcomponents` correctly after running `yarn bootstrap --core`.

Would you mind taking a look?

File: addons/docs/src/frameworks/web-components/custom-elements.ts [hunk 2]
```diff
@@ -39,15 +50,18 @@ function mapData(data: TagItem[], category: string) {
   return (
     data &&
     data.reduce((acc, item) => {
-      const type = category === 'properties' ? { name: item.type } : { name: 'void' };
+      if (item?.kind === 'method') return acc;
```
Comment (gaetanmaisse) [inline addons/docs/src/frameworks/web-components/custom-elements.ts:53]: Can `item` not be defined here? or is the `?` an extra security just to be sure?

**Comment by gaetanmaisse** [comment]
@thepassle I think you were just an export away 😉 

I push a couple of commits to add JSDoc on the component used in the WC kitchen sink, add an NPM Script to run the analyzer easily and it looks like it is working: 
image

On the screenshot you will see that the descriptions are empty for the properties even if I added some in the JSDoc, I think it's because I activated the `--litelement` option and so we are ending with multiple times the same property in the manifest: 
 - First time based on the JSDoc here: https://github.com/storybookjs/storybook/pull/15138/files#diff-c004e78b410c9dff2857b3a2cf9bc7ad4452e1bf234006d7ce0dc500098616f9R44
 - Second time, auto-discovered by the `--litelement` mode?, here: https://github.com/storybookjs/storybook/pull/15138/files#diff-c004e78b410c9dff2857b3a2cf9bc7ad4452e1bf234006d7ce0dc500098616f9R82

I think we should merge these 2 objects describing the same property. Do you think it's something that could/should be changed on `@custom-elements-manifest/analyzer` side?

Notes: I set a specific version of TypeScript using Yarn's resolution because we need to update Yarn before being able to us the latest version, this isn't related to this PR.

File: addons/docs/src/frameworks/web-components/custom-elements.ts (hunk 2 above)
Comment (thepassle) [reply addons/docs/src/frameworks/web-components/custom-elements.ts:53]: In the `'experimental'` version of the schema, `item` wont have a `kind` here, but in the `'1.0.0'` version of the schema, it does, and we want to be sure its not a method

**Comment by thepassle** [comment]
> On the screenshot you will see that the descriptions are empty for the properties even if I added some in the JSDoc, I think it's because I activated the --litelement option and so we are ending with multiple times the same property in the manifest:

Thats odd — that should not be the case. Let me doublecheck this in the analyzer

**Comment by thepassle** [comment]
@gaetanmaisse The duplicate fields appearing the manifest bug should be fixed in `@custom-elements-manifest/analyzer@0.3.1`. I forgot to add a check to see if there was already an existing field with the same name in the manifest. Thanks for pointing it out 😄 

I think we also need to add some docs to document how to use the new version of the manifest, do we need anything else to move forward with this PR?

**Comment by thepassle** [comment]
@gaetanmaisse @shilman I've updated the documentation 🙂

File: examples/web-components-kitchen-sink/package.json [hunk 3]
```diff
@@ -47,26 +48,30 @@
     "@storybook/source-loader": "portal:../../lib/source-loader",
     "@storybook/theming": "portal:../../lib/theming",
     "@storybook/ui": "portal:../../lib/ui",
-    "@storybook/web-components": "portal:../../app/web-components"
+    "@storybook/web-components": "portal:../../app/web-components",
+    "typescript": "4.2.4"
```
Comment (gaetanmaisse) [inline examples/web-components-kitchen-sink/package.json:52]: Set a fixed version to avoid compat' issue with the Yarn 2 version we are using for now

**Review by gaetanmaisse — APPROVED** [review]
LGTM 🚀  

Notes: e2e errors are not related to this PR, see https://github.com/storybookjs/storybook/issues/15411

**Comment by gaetanmaisse** [comment]
Merged! Thanks for the PR @thepassle and let see what could be the next improvements on that topic :)

**Comment by thepassle** [comment]
Awesome! Thanks so much for the help on this 🙂 Do you have any indication on when this will get released?

**Comment by shilman** [comment]
6.4-alpha = release later today
6.4-final = september

see https://github.com/storybookjs/storybook/issues/15355

**Comment by edoardocavazza** [comment]
Hello!

It seems that it does not handle the `privacy` field, so it also process private properties. Is this the correct behaviour or should I open an issue for that?

**Comment by shilman** [comment]
@edoardocavazza new issue please! 🙏

**Comment by jonniebigodes** [comment]
@thepassle one final thing I would like to ask of you. If you don't mind popping into our [Discord Server](https://discord.gg/UUt2PJb) and message me (same username) to follow up on a couple of items.

**Pull request merged**